### PR TITLE
Fe post likes

### PR DESCRIPTION
**Pull request by LukaszBielsko**

### CODE REVIEW

File: frontend/src/authenticated/EditPost.js [hunk 1]
```diff
@@ -28,16 +34,32 @@ const EditPost = props => {
...
+  useEffect(() => {
+    const userCategories = allCategories.filter(category =>
+      categoriesIds.includes(category.id)
+    );
+    const userCategoriesOptions = convertToSelectOptions(
+      userCategories
+    );
+
+    setUserCategoriesOptions(userCategoriesOptions);
+  }, [allCategories, categoriesIds]);
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/EditPost.js]: keep it in one useEffect :-)

File: frontend/src/authenticated/EditPost.js [hunk 2]
```diff
@@ -58,17 +80,22 @@ const EditPost = props => {
...
+    setButtonState(false);
+
     setTitle(event.target.value);
   };
 
   const handleSelect = selectedOptions => {
+    setButtonState(false);
+
     if (!selectedOptions) {
+      setCategoriesIds([]);
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/EditPost.js]: add empty line before return :-)

File: frontend/src/components/Likes.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from "react";
+
```
Comment (MarcinCholewka) [inline frontend/src/components/Likes.js]: ✂️

File: frontend/src/components/Likes.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        setHasLiked(!hasLiked);
+        setLikesNumber(likesNumber - 1);
+      }
+    } else {
+      const response = await handleLike(id, "POST");
+      if (response) {
+        setHasLiked(!hasLiked);
+        setLikesNumber(likesNumber + 1);
+      }
+    }
```
Comment (MarcinCholewka) [inline frontend/src/components/Likes.js]: suggestion :
avoid to use if statement in if statement. Maybe better solution is to use try().catch() with that promise....

File: frontend/src/utils.js [hunk 5]
```diff
@@ -57,6 +57,23 @@ export const deleteData = async (url, data) => {
...
+// LIKES
+export const handleLike = async (postId, method) => {
+  const response = await fetch(`api/posts/${postId}/likes`, {
+    method: `${method}`,
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      "Content-Type": "application/json"
+    }
+  });
+  return response.ok;
```
Comment (MarcinCholewka) [inline frontend/src/utils.js]: add empty line before return statement

File: frontend/src/utils.js [hunk 6]
```diff
@@ -57,6 +57,23 @@ export const deleteData = async (url, data) => {
...
+};
+
+export const checkHasLiked = (likes, userId) => {
+  const likesArray = likes.map(like => like.user_uuid);
+  return likesArray.includes(userId);
```
Comment (MarcinCholewka) [inline frontend/src/utils.js]: add empty line before return statement

File: frontend/src/authenticated/EditPost.js (hunk 1 above)
Comment (LukaszBielsko) [reply frontend/src/authenticated/EditPost.js]: There would be too many variables in dependency array, causing infinite loop. I tried many approaches and this one works.

File: frontend/src/components/Likes.js (hunk 4 above)
Comment (LukaszBielsko) [reply frontend/src/components/Likes.js]: done as requested

**Pull request merged**